sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -267,9 +267,12 @@ def _get_dataproc_cluster_master_url_if_applicable(
               category=DeprecationWarning)
         project_id = (user_pipeline.options.view_as(GoogleCloudOptions).project)
         region = (user_pipeline.options.view_as(GoogleCloudOptions).region)
-        cluster_name = ie.current_env().clusters.default_cluster_name
-        cluster_metadata = MasterURLIdentifier(
-            project_id=project_id, region=region, cluster_name=cluster_name)
+        if not project_id:
+          cluster_metadata = ie.current_env().clusters.default_cluster_metadata
+        else:
+          cluster_name = ie.current_env().clusters.default_cluster_name
+          cluster_metadata = MasterURLIdentifier(
+              project_id=project_id, region=region, cluster_name=cluster_name)
```

Comment:
Can you comment on why this if case is needed?

Reply:
This case was added in the scenario the user first uses the JupyterLab extension to select a default cluster, at this point, we update ib.clusters.default_cluster_metadata to be the MasterURLIdentifier that is used by the selected cluster. Afterwards, if they create a new InteractiveRunner(underlying_runner=FlinkRunner()) pipeline that does not have a Flink master_url and also does not have a Google Cloud project specified, we just assign the default Dataproc cluster to that pipeline.